sky/spot/controller.py

```diff
@@ -140,11 +134,20 @@ def _run(self):
                         failure_type=spot_state.SpotStatus.FAILED,
                         end_time=end_time)
                     break
+                else:
+                    # Unexpected job status received, try to recover
+                    # the cluster to be safe.
+                    logger.info(f'Got unexpected job status: {job_status}. '
+                                'Recovering...')
```

Comment:
Actually, after thinking more about this part. The only possible `job_status` that can get here is `None`. 
That is to say, the controller fails to connect to the cluster, though the cluster is UP. We can move the logging to the place together with the `Cluster is preempted` logging.
```suggestion
                # Although the cluster is healthy, we fail to access the
                # job status try to recover the job (will not restart the
                # cluster, if the cluster is healthy).
                assert job_status is None, job_status
                logger.info('Failed to fetch the job status while the '
                                    'cluster is healthy. Try to recover the job '
                                    '(the cluster will not be restarted).')
```

Reply:
sounds good. just fixed!